Hexastore.Web/Queue/QueueWriter.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Hexastore.Web.EventHubs;
+using Microsoft.Extensions.Logging;
+
+namespace Hexastore.Web.Queue
+{
+    public class QueueWriter : IDisposable
+    {
+        private readonly BlockingCollection<StoreEvent> _queue;
+        private readonly EventReceiver _eventReceiver;
+        private readonly ILogger _logger;
+        private readonly Task _task;
+        private readonly CancellationTokenSource _cts = new CancellationTokenSource();
+
+        public QueueWriter(EventReceiver eventReceiver, ILogger logger)
+        {
+            _queue = new BlockingCollection<StoreEvent>();
+            _eventReceiver = eventReceiver;
+            _logger = logger;
+            _task = StartReader();
+            _task.Start();
+        }
+
+        public int Length => _queue.Count;
+
+        public void Dispose()
+        {
+            _cts.Cancel();
+            _queue.Dispose();
+        }
+
+        public void Send(StoreEvent storeEvent)
+        {
+            _queue.Add(storeEvent);
+        }
+
+        private Task StartReader()
+        {
+            return new Task(() => {
+                while (!_cts.IsCancellationRequested) {
```

Comment:
You can use a bool instead of a CTS here. If you stay with the BlockingCollection I think you would need to pass the CT to Take to get it to actually stop, but right now the token isn't doing much. Also the CTS needs to be disposed which would cause the other thread to blow up.

Reply:
Yes. Correct, I will pass the CTS on to the blocking collection.